### PR TITLE
[MI-3833]: Link channel modal integration

### PR DESCRIPTION
**Pull request by SaurabhSharma-884**
-  Integrate API for linking
- Add Dialog for retry in case of error
- Add refetch to update RHs automatically

### CODE REVIEW

File: webapp/src/components/Dialog/Dialog.component.tsx [hunk 1]
```diff
@@ -7,12 +7,20 @@ import usePluginApi from 'hooks/usePluginApi';
...
+        destructive: false,
+        show: false,
+        primaryButtonText: '',
+        secondaryButtonText: '',
+        isLoading: false,
+        title: '',
+    };
 
-    const handleClose = () => dispatch(closeDialog());
+    const handleClose = () => dispatch(closeDialog(id));
```
Comment (ayusht2810) [inline webapp/src/components/Dialog/Dialog.component.tsx:23]: use useCallback wherever possible

File: webapp/src/components/LinkChannelModal/LinkChannelModal.component.tsx [hunk 2]
```diff
@@ -2,54 +2,121 @@ import React, {useEffect, useState} from 'react';
...
+        }
         onClose();
     };
 
+    const handleChannelLinking = () => {
+        const payload: LinkChannelsPayload = {
+            mattermostTeamID: currentTeamId || '',
+            mattermostChannelID: mMChannel?.id || '',
+            msTeamsTeamID: mSTeam?.ID || '',
+            msTeamsChannelID: mSChannel?.ID || '',
```
Comment (ayusht2810) [inline webapp/src/components/LinkChannelModal/LinkChannelModal.component.tsx:52]:
```suggestion
            mattermostChannelID: mmChannel?.id || '',
            msTeamsTeamID: msTeam?.ID || '',
            msTeamsChannelID: msChannel?.ID || '',
```

File: webapp/src/components/LinkChannelModal/SearchMMChannels/SearchMMChannels.component.tsx [hunk 3]
```diff
@@ -34,20 +40,21 @@ export const SearchMMChannels = ({
...
                 then((channels) => {
-                    const suggestions = [];
-                    for (const channel of channels) {
+                    const suggestions:ListItemType[] = [];
+                    for (const channel of channels as MMChannelTypes.Channel[]) {
                         suggestions.push({
                             label: channel.display_name,
                             value: channel.id,
+                            secondaryLabel: teams[channel.team_id].display_name,
+                            icon: channel.type === 'P' ? 'Lock' : 'Globe',
```
Comment (ayusht2810) [inline webapp/src/components/LinkChannelModal/SearchMMChannels/SearchMMChannels.component.tsx:51]: We have constant present for `p`. Can't we use that here?

File: webapp/src/components/LinkChannelModal/SearchMMChannels/SearchMMChannels.component.tsx [hunk 4]
```diff
@@ -34,20 +40,21 @@ export const SearchMMChannels = ({
         if (searchFor && teamId) {
```
Comment (ayusht2810) [inline webapp/src/components/LinkChannelModal/SearchMMChannels/SearchMMChannels.component.tsx:40]: do we need this `teamId` in the condition now?

File: webapp/src/components/Dialog/Dialog.component.tsx (hunk 1 above)
Comment (SaurabhSharma-884) [reply webapp/src/components/Dialog/Dialog.component.tsx:23]: Modified in later PRs

**Review by ayusht2810 — APPROVED** [review]

File: webapp/src/constants/common.constants.ts [hunk 5]
```diff
@@ -18,3 +16,10 @@ export const alertSeverity: Record<SnackbarColor, SnackbarColor> = {
 } as const;
 
 export const alertTimeout = 4000;
+
+export const debounceFunctionTimeLimit = 300;
```
Comment (avas27JTG) [inline webapp/src/constants/common.constants.ts:20]:
```suggestion
export const debounceFunctionTimeLimitInMilliseconds = 300;
```

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**